src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs

```diff
@@ -0,0 +1,593 @@
+        {
+            if (message == null)
+            {
+                return null;
+            }
+            if (this.SecurityProtocol == null)
+                throw new Exception("SecurityProtocol is null");
+            ThrowIfSecureConversationCloseMessage(message);
+           // using (this.ApplyHostingIntegrationContext(message))
+           // {
+                return this.SecurityProtocol.VerifyIncomingMessage(ref message, timeout, correlationState);
+           // }
+        }
+
+
+        [SecuritySafeCritical]
+        internal void VerifyIncomingMessage(ref Message message, TimeSpan timeout)
+        {
+            if (message == null)
+            {
+                return;
+            }
+            ThrowIfSecureConversationCloseMessage(message);
+           // using (this.ApplyHostingIntegrationContext(message))
+           // {
+                this.SecurityProtocol.VerifyIncomingMessage(ref message, timeout);
+           // }
+        }
+    }
+    class SecurityReplyChannel : ServerSecurityChannel<IReplyChannel>, IReplyChannel
+    {
+        bool sendUnsecuredFaults;
+        internal static readonly SecurityStandardsManager defaultStandardsManager = SecurityStandardsManager.DefaultInstance;
+
+        public SecurityReplyChannel(SecurityServiceDispatcher securityDispatcher, IReplyChannel outerChannel)
+            : base(outerChannel)
+        {
+            this.SecurityServiceDispatcher = securityDispatcher;
+        }
+
+        public EndpointAddress LocalAddress => throw new NotImplementedException();
+
+        public IServiceChannelDispatcher ChannelDispatcher { get; set; }
+
+        public SecurityServiceDispatcher SecurityServiceDispatcher { get; }
+        public CommunicationState State => this.OuterChannel.State;
+       
+        public virtual XmlDictionaryString RenewAction
+        {
+            get
+            {
+                return defaultStandardsManager.SecureConversationDriver.RenewAction;
+            }
+        }
+
+        public virtual XmlDictionaryString IssueAction
+        {
+            get
+            {
+                return defaultStandardsManager.SecureConversationDriver.IssueAction;
+            }
+        }
+
+        public event EventHandler Closed;
+        public event EventHandler Closing;
+        public event EventHandler Faulted;
+        public event EventHandler Opened;
+        public event EventHandler Opening;
+
+        public void Abort()
+        {
+            
+        }
+
+        public Task CloseAsync()
+        {
+           return Task.CompletedTask;
+        }
+
+        public Task CloseAsync(CancellationToken token)
+        {
+            return Task.CompletedTask;
+        }
+
+        public T GetProperty<T>() where T : class
+        {
+            return this.OuterChannel.GetProperty<T>();
+        }
+
+        public Task OpenAsync()
+        {
```

Comment:
Where did the open/close implementations come from? They aren't there on this class in the original implementation.